File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int m_profit=0;

        for(int i=0;i<prices.length-1;i++){
            if(prices[i+1]>prices[i]){
                for(int j=i+1;j<prices.length;j++){
                    if(prices[j]<prices[i]) break;
                    if((prices[j]-prices[i])>m_profit){
                        m_profit=prices[j]-prices[i];
                    }
                }
            }
            
        }
        return m_profit;
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        int sell = prices[0];
        int buy = prices[0];

        int profit = 0;

        for(int i = 0; i < n; i++){
            if(prices[i] < buy){
                buy = prices[i];
                sell = prices[i];
            }
            else if(prices[i] > sell){
                sell = prices[i];
                profit = max(sell - buy, profit);
            }
        }

        return profit;
    }
};